webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -893,16 +917,17 @@ private Stream<List<TextUnitDTO>> partitionSingulars(
       String localeTag,
       String skipTextUnitsWithPattern,
       String skipAssetsWithPathPattern) {
-    return partitionedStream(
+    TextUnitSearcherParameters parameters =
         baseParams(
             repositoryId,
             localeTag,
             skipTextUnitsWithPattern,
             skipAssetsWithPathPattern,
             true,
             true,
-            null),
-        textUnitSearcher::search);
+            null);
+    parameters.setOrdered(true);
```

Comment:
Looking at the name of this param here - I'm thinking it might be worth detailing the order type in the name, e.g. something similar to: setOrderByTextUnitID (as another natural ordering function could by date created/updated etc.).

Also - should this be set as part of baseParams() above? I don't have any strong opinion on it.

Reply:
Side note: long-term batching by a static count + ordered by creation date/TU ID is not ideal - as it's not resistant to deletes and still results in large shifts of everything after that ID when that occurs - even if there are no changes whatsoever in subsequent batches.

The ideal solution long-term (not for this PR) likely involves a fix set of "batches"/buckets where the bucket count correlates to the target repo size - and where we do some form of consistent hashing in those buckets (mod % bucket size) - to ensure a single delete doesn't have a cascading effect on everything added after it's creation time.

Reply:
It was initially in `baseParams()` but tests failed in HSQL at the count query stage, it didn't seem to like the count query with order by id (this wasn't an issue for mysql).

Removing from `baseParams()` fixed the issue.

Happy to discuss changing to a consistent hashing approach!